packages/server/src/fhir/repo.ts

```diff
@@ -1019,6 +1019,13 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
     try {
       const resource = await this.readResourceImpl(resourceType, id);
 
+      if (patch.find((operation) => operation.path === '/resourceType')) {
+        throw new OperationOutcomeError(badRequest('Incorrect Resource Type'));
+      }
+      if (patch.find((operation) => operation.path === '/id')) {
+        throw new OperationOutcomeError(badRequest('Incorrect Id'));
+      }
+
       try {
         const patchResult = applyPatch(resource, patch).filter(Boolean);
```

Comment:
@mattwiller I moved the checks to `updateResourceImpl()`, but my new spec fails with a validation error(`Invalid additional property`) from the `applyPatch` function, which gets called before `updateResourceImpl()`.

Would it make sense to have checks at both levels? One to check that the patch call is valid, and another at the `updateResourceImpl` to prevent the issue in depth